### ISSUE TITLE
refactor storage lib

### ISSUE THREAD

**Issue by rtens** [rtens/xkdl#10]
Reading and writing of tasks needs to be refactored. The current solution is a crutch.

Either use something Repository-like or even pomf-ish

**Comment by cebe** [comment]
would be good to have a kind of abstraction layer over storage so we can implement different storage solutions.

**Comment by rtens** [comment]
Yeah that's the idea behind it. The current abstraction is pretty flaky.
